test/ex_webrtc/data_channel_test.exs

```diff
@@ -0,0 +1,170 @@
+      label1 = "my label"
+      {:ok, %DataChannel{ref: ref1}} = PeerConnection.create_data_channel(pc1, label1)
+      {:ok, _sender} = PeerConnection.add_track(pc1, MediaStreamTrack.new(:video))
+
+      :ok = negotiate(pc1, pc2)
+      :ok = connect(pc1, pc2)
+
+      assert_receive {:ex_webrtc, ^pc2, {:data_channel, %DataChannel{label: ^label1}}}
+      assert_receive {:ex_webrtc, ^pc1, {:data_channel_state_change, ^ref1, :open}}
+
+      # add more tracks and channels and renegotiate
+      {:ok, _sender} = PeerConnection.add_track(pc1, MediaStreamTrack.new(:video))
+      :ok = negotiate(pc1, pc2)
+
+      label2 = "my label 2"
+      {:ok, %DataChannel{ref: ref2}} = PeerConnection.create_data_channel(pc2, label2)
+      assert_receive {:ex_webrtc, ^pc1, {:data_channel, %DataChannel{label: ^label2}}}
+      assert_receive {:ex_webrtc, ^pc2, {:data_channel_state_change, ^ref2, :open}}
+    end
+
+    test "with channel added only in renegotiation" do
+      {:ok, pc1} = PeerConnection.start_link()
+      {:ok, pc2} = PeerConnection.start_link()
+      {:ok, _sender} = PeerConnection.add_track(pc1, MediaStreamTrack.new(:audio))
+      {:ok, _sender} = PeerConnection.add_track(pc1, MediaStreamTrack.new(:video))
+
+      :ok = negotiate(pc1, pc2)
+      :ok = connect(pc1, pc2)
+
+      label = "my label"
+      {:ok, %DataChannel{ref: ref}} = PeerConnection.create_data_channel(pc2, label)
+
+      # NOTICE: in theory, this should be true, but as of now, we perform the SCTP handshake
+      # even if the SDP did not contain any data lines
+      # refute_receive {:ex_webrtc, ^pc1, {:data_channel, _}}
+
+      :ok = negotiate(pc2, pc1)
+
+      assert_receive {:ex_webrtc, ^pc1, {:data_channel, %DataChannel{label: ^label}}}
+      assert_receive {:ex_webrtc, ^pc2, {:data_channel_state_change, ^ref, :open}}
+    end
+  end
+
+  describe "sending data" do
+    setup do
+      {:ok, pc1} = PeerConnection.start_link()
+      {:ok, pc2} = PeerConnection.start_link()
+      {:ok, %DataChannel{ref: ref1}} = PeerConnection.create_data_channel(pc1, "label")
+
+      :ok = negotiate(pc1, pc2)
+      :ok = connect(pc1, pc2)
+
+      assert_receive {:ex_webrtc, ^pc2, {:data_channel, %DataChannel{ref: ref2}}}
+      assert_receive {:ex_webrtc, ^pc1, {:data_channel_state_change, ^ref1, :open}}
+
+      %{pc1: pc1, pc2: pc2, ref1: ref1, ref2: ref2}
+    end
+
+    test "message from initiating peer", %{pc1: pc1, pc2: pc2, ref1: ref1, ref2: ref2} do
+      data1 = <<1, 2, 3>>
+      :ok = PeerConnection.send_data(pc1, ref1, data1)
+      assert_receive {:ex_webrtc, ^pc2, {:data, ^ref2, ^data1}}
+
+      data2 = for i <- 1..1024, into: <<>>, do: <<i>>
+      :ok = PeerConnection.send_data(pc1, ref1, data2)
+      assert_receive {:ex_webrtc, ^pc2, {:data, ^ref2, ^data2}}
+
+      :ok = PeerConnection.send_data(pc1, ref1, <<>>)
+      assert_receive {:ex_webrtc, ^pc2, {:data, ^ref2, <<>>}}
+    end
```

Comment:
Is there any reason why we send so many messages? If yes, then it would be nice to add a comment

Reply:
I wouldn't consider 3 to be _so many_. Each tests a packet with different properties, which is nice considering that empty packets are handled differently. I should've made the big packet even bigger to get over SCTP message size limit.

Reply:
Sorry, I misread it, thought we send 1024 :man_facepalming: